akka-stream-tests/src/test/scala/akka/stream/io/OutputStreamSourceSpec.scala

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.io
+
+import java.io.IOException
+import java.util.concurrent.TimeoutException
+
+import akka.actor.ActorSystem
+import akka.stream.impl.StreamSupervisor.Children
+import akka.stream.impl.{ ActorMaterializerImpl, StreamSupervisor }
+import akka.stream.scaladsl.Keep
+import akka.stream.testkit.Utils._
+import akka.stream.testkit._
+import akka.stream.testkit.scaladsl.TestSink
+import akka.stream.{ ActorMaterializer, ActorMaterializerSettings }
+import akka.util.ByteString
+
+import scala.concurrent.duration._
+import scala.concurrent.{ Await, Future }
+
+class OutputStreamSourceSpec extends AkkaSpec(UnboundedMailboxConfig) {
+  import system.dispatcher
+
+  val settings = ActorMaterializerSettings(system).withDispatcher("akka.actor.default-dispatcher")
+  implicit val materializer = ActorMaterializer(settings)
+
+  val timeout = 300.milliseconds
+  val bytes = "abc".getBytes
+  val byteString = ByteString("abc")
+
+  def expectBlocked[T](f: Future[T]) =
+    the[Exception] thrownBy Await.result(f, timeout) shouldBe a[TimeoutException]
+
+  def expectSuccess[T](f: Future[T], value: T) =
+    Await.result(f, timeout) should be(value)
+
+  "OutputStreamSource" must {
+    "read bytes from OutputStream" in assertAllStagesStopped {
+      val ((outputStream, f), probe) = OutputStreamSource().toMat(TestSink.probe[ByteString])(Keep.both).run
+      val s = probe.expectSubscription()
+
+      outputStream.write(bytes)
+      s.request(1)
+      probe.expectNext(byteString)
+      outputStream.close()
+      probe.expectComplete()
+      Await.result(f, timeout) should be(3)
+    }
+
+    "block flush call until send all buffer to downstream" in assertAllStagesStopped {
+      val ((outputStream, _), probe) = OutputStreamSource().toMat(TestSink.probe[ByteString])(Keep.both).run
+      val s = probe.expectSubscription()
+
+      outputStream.write(bytes)
+      val f = Future(outputStream.flush())
+
+      probe.expectNoMsg(timeout)
```

Comment:
please move this down one line and set the duration to zero (because `expectBlocked` already waits long enough)